### PR TITLE
[FLINK-19894][python] Use iloc for positional slicing instead of direct slicing in from_pandas

### PR DESCRIPTION
**Pull request by HuangXingBo**
## What is the purpose of the change

*This pull request use iloc for positional slicing instead of direct slicing in `from_pandas`*

## Brief change log

  - *use iloc for positional slicing instead of direct slicing in `from_pandas`*

## Verifying this change

This change added tests and can be verified as follows:

*`test_from_pandas_with_float_index` in `test_pandas_conversion.py`*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: flink-python/pyflink/table/tests/test_pandas_conversion.py [hunk 1]
```diff
@@ -132,6 +132,12 @@ def test_from_pandas(self):
                             "1970-01-01 00:00:00.123,[hello, 中文],1,hello,"
                             "1970-01-01 00:00:00.123,[1, 2]"])
 
+    def test_from_pandas_with_float_index(self):
```
Comment (dianfu) [inline flink-python/pyflink/table/tests/test_pandas_conversion.py:135]: Is it possible integrate this test case into an existing test case? Then we could avoid introduce another IT case.

Comment (HuangXingBo) [reply flink-python/pyflink/table/tests/test_pandas_conversion.py]: Yes. Make sense

**Comment by HuangXingBo** [comment]
@dianfu Thanks a lot for the review. I have addressed the comment at the latest commit.

**Review by dianfu — APPROVED** [review]
LGTM.

**Pull request closed without merging**